ambari-server/src/main/java/org/apache/ambari/server/controller/internal/MpackResourceProvider.java

```diff
@@ -159,12 +159,15 @@ public RequestStatus createResourcesAuthorized(final Request request)
         associatedResources.add(resource);
         return getRequestStatus(null, associatedResources);
       }
-    } catch (IOException e) {
-      if (e instanceof ConnectException)
-        throw new SystemException("The Mpack Uri : " + mpackRequest.getMpackUri() + " is not valid. Please try again");
-      e.printStackTrace();
-    } catch (BodyParseException e1) {
-      e1.printStackTrace();
+    }
+    catch (ConnectException e) {
+      throw new SystemException("The Mpack Uri: " + mpackRequest.getMpackUri() + " is not valid. Please try again", e);
```

Comment:
Wouldn't a ConnectException also sometimes occur for a valid URI that happens to not be reachable with the given connection?  

Its not a big deal for now, but maybe we should consider changing the error message here to make the error cause a little clearer.

Reply:
Agreed it could be improved. The reason why I made this change is that the original code was swallowing exceptions and I could only see an NPE thrown elsewhere and had to spend time debugging what was actually happening.